tests/vm_test.go

```diff
@@ -1921,6 +1923,72 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("[Serial] when node becomes unhealthy", Serial, func() {
+		const componentName = "virt-handler"
+		var nodeName string
+		var numberOfVirtHandler int32
+
+		handlerPodGetter := func(virtCli kubecli.KubevirtClient, componentName, namespace, nodeName string) *k8sv1.PodList {
+			pods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(),
+				metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("kubevirt.io=%s", componentName),
+					FieldSelector: fmt.Sprintf("spec.nodeName=" + nodeName),
+				})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pods.Items).To(HaveLen(1))
+
+			return pods
+		}
+
+		AfterEach(func() {
+			tests.BlackHolePod(handlerPodGetter, virtClient, componentName, flags.KubeVirtInstallNamespace, nodeName, tests.BlackHoleActionDel)
+			Eventually(tests.GetReadyDaemonSetReplicas(virtClient, componentName, flags.KubeVirtInstallNamespace),
+				30*time.Second, time.Second).Should(BeNumerically("==", numberOfVirtHandler))
+		})
+
+		It("[Serial] the VMs running in that node should be respawned", func() {
+			By("Starting VM")
+			vm := startVM(virtClient, createVM(virtClient, libvmi.NewAlpine()))
+
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			oldUID := vmi.UID
+
+			By("Blocking virt-handler for updating the VMI")
+			nodeName = vmi.Status.NodeName
+			numberOfVirtHandler = tests.GetDesiredDaemonSetReplicas(virtClient, componentName, flags.KubeVirtInstallNamespace)
+			tests.BlackHolePod(handlerPodGetter, virtClient, componentName, flags.KubeVirtInstallNamespace, nodeName, tests.BlackHoleActionAdd)
+			Eventually(tests.GetReadyDaemonSetReplicas(virtClient, componentName, flags.KubeVirtInstallNamespace),
```

Comment:
How do we know it was exactly the virt-handler on the VMI's node, that went down?

Reply:
You are right! Nice catch. I've changed how the virt-handler readiness check is performed